### PR TITLE
Ansible same as machine option

### PR DESCRIPTION
**Pull request by hililh**

### CODE REVIEW

File: Extensions/Ansible/Src/Tasks/Ansible/ansibleUtils.ts [hunk 1]
```diff
@@ -105,6 +106,35 @@ export function runCommandOnRemoteMachine(command: string, sshClient: any, optio
...
+    if (!options) {
+        tl.debug('Options not passed to runCommandOnRemoteMachine, setting defaults.');
+        var options = new RemoteCommandOptions();
+        options.failOnStdErr = true;
+    }
+
+    var cmdToRun = command;
+    tl.debug('cmdToRun = ' + cmdToRun);
+
+    shell.exec(cmdToRun, (err, stdout, stderr) => {
```
Comment (sachinma) [inline Extensions/Ansible/Src/Tasks/Ansible/ansibleUtils.ts]: consider using task-lib's toolrunner

File: Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts [hunk 2]
```diff
@@ -1,204 +1,342 @@
...
-    constructor() {
+export class ansibleCommandLineInterface extends ansibleInterface {
+    constructor(params: ansibleTaskParameters) {
         super();
+
+        this._taskParameters = params;
+        this._remoteCmdOptions = new RemoteCommandOptions();
+    }
+
+    public static getInstance(params: ansibleTaskParameters): ansibleCommandLineInterface {
```
Comment (sachinma) [inline Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts]: move it to factory and create separate files for each class

File: Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts [hunk 3]
```diff
@@ -1,204 +1,342 @@
...
+        }
+        if (params.ansibleInterface == 'agentMachine') {
+            return new ansibleAgentMachineInterface(params);
+        }
     }
 
     public async execute() {
         this.sshRun();
     }
     private async sshRun() {
```
Comment (sachinma) [inline Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts]: rename to something else

File: Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts (hunk 2 above)
Comment (hililh) [inline Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts]: Will do in next iteration

File: Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts (hunk 3 above)
Comment (hililh) [inline Extensions/Ansible/Src/Tasks/Ansible/ansibleCommandLineInterface.ts]: Will do in next iteration

File: Extensions/Ansible/Src/Tasks/Ansible/ansibleUtils.ts (hunk 1 above)
Comment (hililh) [inline Extensions/Ansible/Src/Tasks/Ansible/ansibleUtils.ts]: Will do in next iteration

**Pull request merged**